eng/linkerTests/linkerCorrectnessTests.targets

```diff
@@ -0,0 +1,109 @@
+      <TestConsoleAppSourceFiles>
+        <ProjectFile>$([MSBuild]::NormalizePath('%(ProjectDir)', 'project.csproj'))</ProjectFile>
+        <TestCommand>$([MSBuild]::NormalizePath('%(ProjectDir)', 'bin', '$(Configuration)', 'net5.0', '%(TestRuntimeIdentifier)', 'publish', 'project'))</TestCommand>
+        <TestExecutionDirectory>$([MSBuild]::NormalizeDirectory('%(ProjectDir)', 'bin', '$(Configuration)', 'net5.0', '%(TestRuntimeIdentifier)', 'publish'))</TestExecutionDirectory>
+        <RuntimePackDirectory>$([MSBuild]::NormalizeDirectory('$(ArtifactsBinDir)', 'microsoft.netcore.app.runtime.%(TestRuntimeIdentifier)', '$(Configuration)'))</RuntimePackDirectory>
+      </TestConsoleAppSourceFiles>
+    </ItemGroup>
+
+    <ItemGroup>
+      <TestConsoleApps Include="@(TestConsoleAppSourceFiles->'%(ProjectFile)')">
+        <ProjectCompileItems>%(Identity)</ProjectCompileItems>
+      </TestConsoleApps>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="GenerateProjects"
+          DependsOnTargets="GetTestConsoleApps;CreateTestDir"
+          Inputs="@(TestConsoleAppSourceFiles);$(ProjectTemplate);@(TestSupportFiles)"
+          Outputs="%(TestConsoleApps.Identity)">
+    <PropertyGroup>
+      <_projectDir>%(TestConsoleApps.ProjectDir)/</_projectDir>
+      <_projectFile>%(TestConsoleApps.ProjectFile)</_projectFile>
+      <_projectSourceFile>%(TestConsoleApps.ProjectCompileItems)</_projectSourceFile>
+    </PropertyGroup>
+
+    <MakeDir Directories="$(_projectDir)" />
+    <WriteLinesToFile File="$(_projectFile)"
+                      Lines="$([System.IO.File]::ReadAllText('$(ProjectTemplate)').Replace('{RuntimePackDir}', '%(TestConsoleApps.RuntimePackDirectory)').Replace('{TargetingPackDir}','$(MicrosoftNetCoreAppRefPackDir)').Replace('{RuntimeIdentifier}','%(TestConsoleApps.TestRuntimeIdentifier)'))"
+                      Overwrite="true" />
+    <Copy SourceFiles="$(_projectSourceFile)"
+          DestinationFolder="$(_projectDir)" />
+    <Message Text="Generated $(_projectFile)" />
+  </Target>
+
+  <Target Name="RestoreProjects"
+          DependsOnTargets="GenerateProjects">
+    <PropertyGroup>
+      <TestRestoreCommand>"$(TestDotNetPath)"</TestRestoreCommand>
+      <TestRestoreCommand>$(TestRestoreCommand) restore</TestRestoreCommand>
+      <TestRestoreCommand>$(TestRestoreCommand) /nr:false</TestRestoreCommand>
+      <TestRestoreCommand>$(TestRestoreCommand) /warnaserror</TestRestoreCommand>
+      <TestRestoreCommand>$(TestRestoreCommand) -p:configuration=Release</TestRestoreCommand>
+    </PropertyGroup>
+
+    <Exec Command="$(TestRestoreCommand)" StandardOutputImportance="Low" WorkingDirectory="%(TestConsoleApps.ProjectDir)" />
+  </Target>
+
+  <Target Name="PublishTrimmedProjects"
+          DependsOnTargets="RestoreProjects">
+    <PropertyGroup>
+      <TestRestoreCommand>"$(TestDotNetPath)"</TestRestoreCommand>
+      <TestRestoreCommand>$(TestRestoreCommand) msbuild /t:PublishTrimmed</TestRestoreCommand>
```

Comment:
same question here, why aren't you using the msbuild task instead of an out-of-proc dotnet invocation?

Reply:
Here we do out of proc because we don't want any of the properties that are on the current msbuild session to propagate to the test projects, we do the same for packaging projects for the same reason. I did however took your advise and moved the Restore out-of-proc and moved it to happen in-proc on the PublishTrimmed target.

Reply:
You can specify the props that you don't want to be propagated in the RemoveProperties attribute, also you can specify the props that you explicitly want to pass into the target via the Properties attribute. Are you sure the limitation applies?

Reply:
While it is true that you could RemoveProperties, global properties that we use to build runtime changes over time, so that means that even if we get this right today, it is possible in the future we set a different one that will flow down to these tests and for this ones we really want to be in as close to an isolated environment as possible. Similar to the package tests, we want to emulate users projects which won't flow down any of the state or infra that this repo uses, so I still feel that running out of proc is what we want here.